tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4207,6 +4210,135 @@ TEST_F(ValidationTest, ValidateWithHash) {
   VERIFY_ARE_EQUAL(memcmp(Result, pHeader->Hash.Digest, sizeof(Result)), 0);
 }
 
+std::wstring string_to_wstring(const std::string &str) {
+  if (str.empty())
+    return L"";
+
+  int size_needed =
+      MultiByteToWideChar(CP_UTF8, 0, str.c_str(), (int)str.size(), nullptr, 0);
+  std::wstring wstrTo(size_needed, 0);
+  MultiByteToWideChar(CP_UTF8, 0, str.c_str(), (int)str.size(), &wstrTo[0],
+                      size_needed);
+  return wstrTo;
+}
+
+std::string wstring_to_utf8(const std::wstring &wstr) {
+  if (wstr.empty())
+    return "";
+
+  int size = WideCharToMultiByte(CP_UTF8, 0, wstr.data(), -1, nullptr, 0,
+                                 nullptr, nullptr);
+  std::string result(size - 1, 0); // -1 to exclude null terminator
+  WideCharToMultiByte(CP_UTF8, 0, wstr.data(), -1, &result[0], size, nullptr,
+                      nullptr);
+  return result;
+}
+
+std::string GetEnvVarA(const std::string &varName) {
+#ifdef _WIN32
+  DWORD size = GetEnvironmentVariableA(varName.c_str(), nullptr, 0);
```

Comment:
`GetEnvironmentVariableA` gets an ANSI string, not a UTF-8 encoded Unicode string.  Should use `GetEnvironmentVariableW` instead.  The test should then just use the wide string for getting and setting environment variables, to ensure they are Unicode.  This will result in more string translation in the test if not using separate paths for WIN32/*nix ,but the cost will be negligable, similar to how we must use wide strings in arguments in the DXC APIs.

Alternatively, we could just use `std::getenv` and `_putenv_s` on WIN32.  Does `std::getenv` not work as expected on WIN32?